### PR TITLE
Remove class signature lines from Line Number Table

### LINKED ISSUE #32730: Class/service signature lines are getting added as instruction lines during codegen
**Description:**
Please note the $subject. Due to the above reason, the debugger provides unexpected outputs in similar contexts (e.g. in the below program, when trying to stepping into object init method from the main function, debugger navigates to the first line of the class definition instead of hitting `init` method). Therefore we might need to ignore the first(signature) and last (right brace) lines of class/service definitions from source line mappings during codegen.  

Note: A similar issue is reported in #30740, which will be easier to reproduce.

```ballerina
class Person { // this line shouldn't be a valid debug point
    public int age;
    public string firstName;
    public string lastName;

    function init(int age, string firstName, string lastName) {
        self.age = age;
        self.firstName = firstName;
        self.lastName = lastName;
    }

    function getFullName() returns string {
        return self.firstName + " " + self.lastName;
    }
}  // this line shouldn't be a valid debug point

public function main() {
    Person p1 = new (5, "John", "Doe");
    string fullName = p1.getFullName();
}
```
**Steps to reproduce:**
Consider the above code segment:

- Add debug point for the line `Person p1 = new (5, "John", "Doe");` and start the program in debug mode.
- Once the debug point gets hit, send a `STEP_IN` request.
- Debugger hits the first line of `class Person` instead of hitting the `init` function.

**Related Issues (optional):**
https://github.com/ballerina-platform/ballerina-lang/issues/30740

### PR DESCRIPTION
**Pull request by gabilang**
Fixes #32730

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 1]
```diff
@@ -676,7 +677,7 @@ private void addUserDefinedModuleInitInvocationAndReturn(BLangPackage pkgNode) {
         userDefInitInvocation.requiredArgs = Collections.emptyList();
 
         BLangReturn returnStmt = (BLangReturn) TreeBuilder.createReturnNode();
-        returnStmt.pos = pkgNode.initFunction.pos;
+//        returnStmt.pos = pkgNode.initFunction.pos;
```
Comment (Nadeeshan96) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:680]: Shall we remove this commented code line, if it has no specific purpose

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 2]
```diff
@@ -351,6 +351,7 @@
     private static final String GENERATED_ERROR_VAR = "$error$";
     private static final String HAS_KEY = "hasKey";
     private static final String CREATE_RECORD_VALUE = "createRecordFromMap";
+    private static final String METHOD_INIT = "$init$";
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:354]: This should be defined as constant in Names. Can u check?

Comment (gabilang) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java]: Addressed with https://github.com/ballerina-platform/ballerina-lang/pull/32838/commits/f202d01ccb60a0671bfcc61d3c9387e9e1b16e4e

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/DebugInstructionTest.java [hunk 3]
```diff
@@ -108,9 +108,6 @@ public void objectDebugInstructionTest() throws BallerinaTestException {
         // debugHitInfo = debugTestRunner.waitForDebugHit(10000);
         // Assert.assertEquals(debugHitInfo.getLeft(), new BallerinaTestDebugPoint(debugTestRunner.testEntryFilePath,
         // 33));
-        debugTestRunner.resumeProgram(debugHitInfo.getRight(), DebugTestRunner.DebugResumeKind.STEP_OVER);
```
Comment (warunalakshitha) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/DebugInstructionTest.java:111]: Why do we need to remove this?Can you check with @NipunaRanasinghe

Comment (gabilang) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/DebugInstructionTest.java:111]: According to the requirement the last line (right brace) of the class definition also needed to be removed from the line number table. Previously that line was hit when we do step over or step in and removed with the new change. So the line#27 in the test not going to be hit by debug pointer.

Comment (NipunaRanasinghe) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/DebugInstructionTest.java:111]: @warunalakshitha @gabilang yes thats correct. Need to remove L111-L113 as the list line wont be hit, and need to uncomment L103-L110 that was disabled due to this issue

**Comment by warunalakshitha** [comment]
Closing this one since I have merge the other pr to beta 3 branch.

**Pull request closed without merging**